devito/function_manager.py

```diff
@@ -46,10 +52,20 @@ def generate_function_signature(self, function_descriptor):
             function_params = function_params + [param_vec_def]
         if self.mic_flag:
             function_params += [cgen.Pointer(cgen.POD(type_label, name+"_pointer")) for type_label, name in function_descriptor.value_params]
-            return cgen.FunctionDeclaration(cgen.Value(self._pymic_attribute + '\nint', function_descriptor.name),
+
+            # sets function return type if specified
+            return_type = function_descriptor.return_type if function_descriptor.return_type else self._pymic_attribute + '\nint'
+
+            return cgen.FunctionDeclaration(cgen.Value(return_type, function_descriptor.name),
                                             function_params)
         else:
             function_params += [cgen.POD(type_label, name) for type_label, name in function_descriptor.value_params]
+
+            # used in auto tune
+            if function_descriptor.return_type:  # sets return type if specified
+                return cgen.FunctionDeclaration(cgen.Value(function_descriptor.return_type, function_descriptor.name),
```

Comment:
- FunctionDescriptor can have a parameter that sets whether an `extern` qualifier is needed for the function declaration. Use that instead of `return_type` to decide whether to add `extern` or not. 
- Set `int` as the default value for `FunctionDescriptor`'s `return_type` (in the class, not here). Use that instead of hardcoding `int` here.